extension/src/standalone_pages/InstructorDashboardRenderVideoInfo.js

```diff
@@ -194,6 +220,50 @@ function renderPassiveFeedback(video, response) {
             }
         }
     });
+
+    let keywordsMap = new Map();
+
+    labels.forEach((timestamp) => {
+        getVideoContextKeyWords(keywordsMap, video, timestamp);
```

Comment:
this approach seems good for now but will likely fall apart for the purpose of demonstration. For the recorded demo, we will most likely inject a lot of data points in the passive graph. The amount of network calls and the total cost to retrieve all these information will be very high.

Also, the current setup (this loop and the dependency functions) favors rendering over data. You are not guaranteed to have the data before adding the on-click. This is obviously not wrong but introduces side-effects under the hood for a long duration of time. Coupled with the amount of calls, the amount of time for which data will be in-transit mode is very high.

Now, it seems rather clear to me that why you want to render the fake loading modal. To be honest, the execution flow is somewhat convoluted and comes with caveats (side-effects) under the hood. Not to mention, UI experience deteriorates drastically.

I would like to propose a different execution code in which keywords are being extracted on-demand for every click. If you are worried about redundant API calls, you can leverage `keywordsMap` as your client side cache.

The wait-time per call is fine for now. If you worried about the experience/flow. You can add a one-liner to modify the cursor as shown [here](https://www.w3schools.com/jsref/prop_style_cursor.asp) (low effort but great UI/UX improvement). You will then toggle the cursor change as needed when the request is in-transit in `displayContextKeyWords`.